lib/literal/enum.rb

```diff
@@ -1,15 +1,16 @@
 # frozen_string_literal: true
 
-TracePoint.trace(:end) do |tp|
-	it = tp.self
+class Literal::Enum < Literal::Struct
```

Comment:
It might make sense to extend `Literal::DataStructure` rather than `Literal::Struct` here as we definitely want different marshalling behaviour.